src/core/core.scale.js

```diff
@@ -651,8 +653,12 @@ module.exports = function(Chart) {
 			// Draw all of the tick labels, tick marks, and grid lines at the correct places
 			helpers.each(itemsToDraw, function(itemToDraw) {
 				if (gridLines.display) {
+					context.save();
 					context.lineWidth = itemToDraw.glWidth;
 					context.strokeStyle = itemToDraw.glColor;
+					if (itemToDraw.glBorderDash) {
```

Comment:
this check should also check for the existence of `context.setLineDash`. On IE 9 and 10 this does not exist.